vcx/wrappers/node/src/vcx.ts

```diff
@@ -18,14 +19,21 @@ export class VCXRuntime {
 
   constructor (config: IVCXRuntimeConfig = {}) {
     this._config = config
-     // initialize FFI
+        // initialize FFI
     const libraryPath = this._initializeBasepath()
     this.ffi = ffi.Library(libraryPath, FFIConfiguration)
   }
 
   private _initializeBasepath = (): string => {
-    const library = 'libvcx.so' // TODO: FIXME provide better way to resolve library
+    const platform = os.platform()
+    const postfix = (platform === 'darwin') ? 'dylib' : (platform === 'win32') ? 'dll' : 'so'
+    const library = `libvcx.${postfix}`
+    const libDir = (platform === 'darwin')
+        ? '/usr/local/lib/'
+        : (platform === 'win32')
+            ? 'c:\\windows\\system32\\'
+            : '/usr/lib/'
```

Comment:
I suggest using dictionaries for extensions and paths like:
`extention = {"darwin": ".dylib", "linux": ".so", "win32": ".dll"}`

Reply:
changed